waku/v2/node/peer_manager/waku_peer_store.nim

```diff
@@ -20,74 +25,107 @@ type
+    # Extended custom fields
+    connection*: Connectedness
+    disconnectTime*: int64
+    origin*: PeerOrigin
 
-proc new*(T: type WakuPeerStore): WakuPeerStore =
-  let
-    addressBook = AddressBook(book: initTable[PeerID, seq[MultiAddress]]())
-    protoBook = ProtoBook(book: initTable[PeerID, seq[string]]())
-    keyBook = KeyBook(book: initTable[PeerID, PublicKey]())
-    connectionBook = ConnectionBook(book: initTable[PeerID, Connectedness]())
-    disconnectBook = DisconnectBook(book: initTable[PeerID, int64]())
-  
-  T(addressBook: addressBook,
-    protoBook: protoBook,
-    keyBook: keyBook,
-    connectionBook: connectionBook,
-    disconnectBook: disconnectBook)  
-
-#####################
-# Utility functions #
-#####################
-
-proc add*[T](peerBook: SeqPeerBook[T],
-             peerId: PeerId,
-             entry: T) =
-  ## Add entry to a given peer. If the peer is not known,
-  ## it will be set with the provided entry.
-  
-  peerBook.book.mgetOrPut(peerId,
-                          newSeq[T]()).add(entry)
-  
-  # TODO: Notify clients?
-
-##################  
+##################
 # Peer Store API #
 ##################
 
-proc get*(peerStore: WakuPeerStore,
+proc get*(peerStore: PeerStore,
           peerId: PeerID): StoredInfo =
   ## Get the stored information of a given peer.
-  
   StoredInfo(
+    # Taken from nim-libp2
     peerId: peerId,
-    addrs: peerStore.addressBook[peerId],
-    protos: peerStore.protoBook[peerId],
-    publicKey: peerStore.keyBook[peerId]
+    addrs: peerStore[AddressBook][peerId],
+    protos: peerStore[ProtoBook][peerId],
+    publicKey: peerStore[KeyBook][peerId],
+    agent: peerStore[AgentBook][peerId],
+    protoVersion: peerStore[ProtoVersionBook][peerId],
+
+    # Extended custom fields
+    connection: peerStore[ConnectionBook][peerId],
+    disconnectTime: peerStore[DisconnectBook][peerId],
+    origin: peerStore[SourceBook][peerId],
   )
 
-proc peers*(peerStore: WakuPeerStore): seq[StoredInfo] =
+proc peers*(peerStore: PeerStore): seq[StoredInfo] =
   ## Get all the stored information of every peer.
-  
-  let allKeys = concat(toSeq(keys(peerStore.addressBook.book)),
-                       toSeq(keys(peerStore.protoBook.book)),
-                       toSeq(keys(peerStore.keyBook.book))).toHashSet()
+  let allKeys = concat(toSeq(peerStore[AddressBook].book.keys()),
+                       toSeq(peerStore[ProtoBook].book.keys()),
+                       toSeq(peerStore[KeyBook].book.keys())).toHashSet()
 
   return allKeys.mapIt(peerStore.get(it))
+
+proc peers*(peerStore: PeerStore, proto: string): seq[StoredInfo] =
+  # Return the known info for all peers registered on the specified protocol
+  peerStore.peers.filterIt(it.protos.contains(proto))
+
+proc peers*(peerStore: PeerStore, protocolMatcher: Matcher): seq[StoredInfo] =
+  # Return the known info for all peers matching the provided protocolMatcher
+  peerStore.peers.filterIt(it.protos.anyIt(protocolMatcher(it)))
```

Comment:
This proc's name is a "noun". We use nouns for accessing properties (e.g., a property that we did not make public because we don't want to allow modifying it) or "dynamic properties" (virtual properties that are derived from other properties). These are functions as they have parameters. Please, consider renaming the methods to something like `getPeersByProtocol` or something similar.

Reply:
+1 `getPeersByProtocol`. this proc already existed, so just reused the name. will followup with a pr with renamings once the heavy stuff is done.